resource_type.go

```diff
@@ -134,7 +134,7 @@ func (t ResourceType) validatePatch(r *http.Request) ([]PatchOperation, *errors.
 
 	// The body of each request MUST contain the "schemas" attribute with the URI value of
 	// "urn:ietf:params:scim:api:messages:2.0:PatchOp".
-	if len(req.Schemas) != 1 && req.Schemas[0] != "urn:ietf:params:scim:api:messages:2.0:PatchOp" {
+	if len(req.Schemas) < 1 || req.Schemas[0] != "urn:ietf:params:scim:api:messages:2.0:PatchOp" {
```

Comment:
Is there a reason to use `<` over `!=`?

Reply:
As in the comments, != 1 means that test passes if len(req.Schemas) == 0, which means that the next clause of the && operator is executed.  When req.Schemas length is 0 that causes a panic.  You could say if len(req.Schemas) == 0 || ...,  that's a matter of taste I suppose.

Reply:
Yet, if the `len(req.Schemas)` is `2` that the request is also invalid.

Reply:
So why not just:
```
len(req.Schemas) != 1 || req.Schemas[0]
```

Reply:
Pushed the requested change